src/api/c/binary.cpp

```diff
@@ -55,6 +60,36 @@ static inline af_array arithOp(const af_array lhs, const af_array rhs,
     return getHandle(arithOp<T, op>(l, r, odims));
 }
 
+template<typename T, af_op_t op>
+static inline af_array arithOpBroadcast(const af_array lhs,
+                                        const af_array rhs) {
+    const ArrayInfo &linfo = getInfo(lhs);
+    const ArrayInfo &rinfo = getInfo(rhs);
+
+    dim4 odims(1), ltile(1), rtile(1);
+    dim4 lshape = linfo.dims();
+    dim4 rshape = rinfo.dims();
+
+    for (int d = 0; d < AF_MAX_DIMS; ++d) {
+        DIM_ASSERT(
+            1, ((lshape[d] == rshape[d]) || (lshape[d] == 1 && rshape[d] > 1) ||
+                (lshape[d] > 1 && rshape[d] == 1)));
+        odims[d] = std::max(lshape[d], rshape[d]);
+        if (lshape[d] == rshape[d]) {
+            ltile[d] = rtile[d] = 1;
+        } else if (lshape[d] == 1 && rshape[d] > 1) {
+            ltile[d] = odims[d];
+        } else if (lshape[d] > 1 && rshape[d] == 1) {
+            rtile[d] = odims[d];
+        }
+    }
+
+    Array<T> lhst = tile<T>(modDims(getArray<T>(lhs), lshape), ltile);
```

Comment:
This operation to tile will always call the kernel even if the tile operation could be performed by the JIT.

It would be better to move the tile operation to the common namespace. I have a commit that does this that I will be pushing soon.